tests/testthat/test-util-checkSpec.R

```diff
@@ -58,3 +58,35 @@ test_that("Multiple missing columns are correctly reported", {
     regexp = "missing columns are: df1\\$b and df2\\$y"
   )
 })
+
+test_that("Missing column only gets a flag when it is required", {
+  lData <- list(reporting_groups = gsm::reportingGroups)
+  lSpec <- list(
+    reporting_groups = list(
+      GroupID = list(required = TRUE),
+      GroupLevel = list(required = TRUE),
+      Param = list(required = TRUE),
+      Value = list(required = TRUE),
+      NewVar = list(required = FALSE)
+    )
+  )
+  expect_message(
+    CheckSpec(lData, lSpec),
+    regexp = "All 4 required columns"
+  )
+
+  lSpec <- list(
+    reporting_groups = list(
+      GroupID = list(required = TRUE),
+      GroupLevel = list(required = TRUE),
+      Param = list(required = TRUE),
+      Value = list(required = TRUE),
+      NewVar = list(required = TRUE)
+    )
+  )
+  expect_message(
```

Comment:
(Part of an ongoing conversation, I think)

Shouldn't this throw an error? If the columns are required and missing, isn't that an error? If not... what does "required" mean?

I think the message is wrong, too, but @lauramaxwell is probably working on that.

Reply:
Remaining piece of this is addressed in #1678, we can deal with it later.

Reply:
yep - fixed this in checkSpec to get the message to match correctly. not sure how that got lost